### DISCUSSION TITLE
Is it possible to insert JSON directly?

### DISCUSSION THREAD

**Discussion by hacel** [ClickHouse/ch-go#417] [Q&A] [answered]
Without parsing or unmarshalling it that is. Possible?

**Comment by SpencerTorres** [comment, marked as answer]
We have a new JSON type, and I recently added support for JSON strings in #1034

**Comment by hacel** [comment]
@SpencerTorres Actually I may have misunderstood your commit, I don't think this is useful for my use case. I want to replace the Kafka table engine with a Go consumer but as JSON decoding is not very fast in Go, I would rather let ClickHouse handle that. And there's also the pain of having another place where I have to specify my data schema. I want something like how you would insert a file full of JSON, but through Go. Is that possible?
